client/state/sites/actions.js

```diff
@@ -119,23 +119,29 @@ export function requestSite( siteId ) {
 			siteId,
 		} );
 
-		return wpcom
-			.site( siteId )
-			.get()
-			.then( site => {
-				dispatch( receiveSite( omit( site, '_headers' ) ) );
-				dispatch( {
-					type: SITE_REQUEST_SUCCESS,
-					siteId,
-				} );
-			} )
-			.catch( error => {
-				dispatch( {
-					type: SITE_REQUEST_FAILURE,
-					siteId,
-					error,
+		return new Promise( ( resolve, reject ) => {
```

Comment:
Ahhh, I see! Thanks. I think we should pass the returned `site` so it is more flexible, not just for our usage in `siteSelection`.

Reply:
I finally went with the approach of keeping it untouched. The problem I was seeing was with tests and other usages of `requestSite`: if you didn't provide another `.catch` after that `.then`, it would spill the error message into console. I'm not sure how to circumvent that since Promises `reject` is expected to be handled with `catch`.

Reply:
Hmm, that's strange: if the `siteRequest` promise is rejected with error, there is always a `catch` handler in the action thunk. You don't need to provide a second one in your app code.

> it would spill the error message into console

Did you get a warning about unhandled promise rejection, or something else?

If you don't want to spend more time with this, then go ahead, but I'm suprised that the approach suggested by me wouldn't work -- I used it many times before.